### ISSUE TITLE
[Spec: JPA Service] Upgrade to Jakarta namespace

### ISSUE THREAD

**Issue by maho7791** [osgi/osgi#552]
I would suggest to upgrade the JPA Service specification to Jakarta namespace and latest JPA version 3.1

The move from 2.x to 3.0 contained just the namespace change from javax to jakarta [1]
The latest release 3.1 contains, from my point of view, no specification relevant changes [2]

[1] https://thorben-janssen.com/migrating-jpa-2-x-to-3-0/
[2] https://projects.eclipse.org/projects/ee4j.jpa/releases/3.1

**Comment by timothyjward** [comment]
This will involve moving the JPA spec to version 2 and renaming the package/spec as we did with REST.

We should also update Transaction Control to 1.1, including a Jakarta persistence resource provider. This may involve splitting the Transaction Control API project so that we can avoid a hard dependency on the old JPA API in the API jars.

**Comment by laeubi** [comment]
@maho7791 any updates on this? I also just recently noted that there was a move from javax->jakarta here

**Comment by laeubi** [comment]
> This will involve moving the JPA spec to version 2 and renaming the package/spec as we did with REST.

Maybe just simply name it `org.osgi.service.jpa3`  or `org.osgi.service.jakarta.jpa`?

**Comment by maho7791** [comment]
Like discussed in the F2F in Karlsruhe we use `org.osgi.service.jakartapersistence`
The new spec number is 161

**Comment by laeubi** [comment]
> Like discussed in the F2F in Karlsruhe we use `org.osgi.service.jakartapersistence` The new spec number is 161

As I was not part of the discussion its hard to guess the reasons but for the jakarta webservice and jakarta websocket we previously agreed on using `org.osgi.service.<original namespace>` what would be `org.osgi.service.jakarta.persistence` (with additional dot).

Also note that chapter 161 is currently already used by the [websocket-spec-draft](https://github.com/osgi/osgi/pull/744/files#diff-ac6b6c8de6ddeb606eac6039ae2550767b514e4999cb80364d0f905c6f55d8a4).

**Comment by maho7791** [comment]
@laeubi Thanks for this comment.
I am open regarding the naming. My personal preference would indeed go to `org.osgi.service.jakarta.persistence`
Then I reserve chapter 162 for Jakarta Persistene

@timothyjward @stbischof what do you say?

**Comment by timothyjward** [comment]
> > As I was not part of the discussion its hard to guess the reasons but for the jakarta webservice and jakarta websocket we previously agreed on using `org.osgi.service.<original namespace>` what would be `org.osgi.service.jakarta.persistence` (with additional dot).

> My personal preference would indeed go to org.osgi.service.jakarta.persistence

We discussed avoiding this sort of naming because the OSGi build makes all sorts of assumptions based on package names. Historical attempts at this such as `org.osgi.service.transaction.control` have caused big headaches and I would prefer not to go through that again.

This is the main reason the Servlet whiteboard ended up as `org.osgi.service.servlet` and typed events are `org.osgi.sevice.typedevent`. I would suggest omitting Jakarta from both package names, so there is no risk that the build will decide they’re both part of a single Jakarta service, or omitting the final dot so it is always `jakartaxyz`.

**Comment by laeubi** [comment]
> We discussed avoiding this sort of naming because the OSGi build makes all sorts of assumptions based on package names. Historical attempts at this such as org.osgi.service.transaction.control have caused big headaches and I would prefer not to go through that again.

Could you explain a bit more what are the problem? I didn't notice any such problems elsewhere with a more "deeper" package name. Also I don't think issues in the build should dictate the used package names this sounds really odd.

**Comment by timothyjward** [comment]
> Could you explain a bit more what are the problem?

If I remember correctly the signature tests started misbehaving, as did some of the packaging and validation steps in the gradle build which started treating the “overlapping” packages as different parts of a single super-spec. This is why `org.osgi.service.event.typed` had to be abandoned and `org.osgi.service.transaction.control` would have been except for the removal of the transactions spec from the compendium booklet.

> Also I don't think issues in the build should dictate the used package names

We do have an ongoing process to get the build updated, but it’s not a small job and the best person for the job no longer has paid time to work on OSGi. 

There are other considerations, for example Jakarta is a trademark which is managed by a different working group, and we don’t (and never intend to) support the whole Jakarta specification set. Having `org.osgi.service.jakarta.*` naming rules risks confusion about both of those things.

Consistency, particularly for a websocket whiteboard, would be best served by `org.osgi.service.websocket` which matches the naming for the Servlet whiteboard, which itself is a close match in terms of goals.

in summary, you asked for reasoning. There were multiple small headaches/concerns with using `org.osgi.service.jakarta.xyz` which is why an alternative was agreed at the F2F.

**Comment by laeubi** [comment]
> in summary, you asked for reasoning. There were multiple small headaches/concerns with using `org.osgi.service.jakarta.xyz` which is why an alternative was agreed at the F2F.

So consequently this one should the be `org.osgi.service.persistence` (the 'old' one was `org.osgi.service.jpa`) ?
